libp2p/protocol_muxer/multiselect_communicator.py

```diff
@@ -1,35 +1,47 @@
-from libp2p.typing import NegotiableTransport
+from libp2p.network.connection.raw_connection_interface import IRawConnection
+from libp2p.stream_muxer.abc import IMuxedStream
+from libp2p.stream_muxer.mplex.utils import decode_uvarint_from_stream, encode_uvarint
+from libp2p.typing import StreamReader
 
 from .multiselect_communicator_interface import IMultiselectCommunicator
 
 
-class MultiselectCommunicator(IMultiselectCommunicator):
-    """
-    Communicator helper class that ensures both the client
-    and multistream module will follow the same multistream protocol,
-    which is necessary for them to work
-    """
+def delim_encode(msg_str: str) -> bytes:
+    msg_bytes = msg_str.encode()
+    varint_len_msg = encode_uvarint(len(msg_bytes) + 1)
+    return varint_len_msg + msg_bytes + b"\n"
 
-    reader_writer: NegotiableTransport
 
-    def __init__(self, reader_writer: NegotiableTransport) -> None:
-        """
-        MultistreamCommunicator expects a reader_writer object that has
-        an async read and an async write function (this could be a stream,
-        raw connection, or other object implementing those functions)
-        """
-        self.reader_writer = reader_writer
+async def delim_read(reader: StreamReader, timeout: int = 10) -> str:
+    len_msg = await decode_uvarint_from_stream(reader, timeout)
+    msg_bytes = await reader.read(len_msg)
+    return msg_bytes.decode().rstrip()
+
+
+class RawConnectionCommunicator(IMultiselectCommunicator):
+    conn: IRawConnection
+
+    def __init__(self, conn: IRawConnection) -> None:
+        self.conn = conn
+
+    async def write(self, msg_str: str) -> None:
+        msg_bytes = delim_encode(msg_str)
+        self.conn.writer.write(msg_bytes)
```

Comment:
Should this be awaited?

Reply:
I don't think so. `conn.writer` is `asyncio.StreamWriter`, and `asyncio.StreamWriter.write` is a sync function.